File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1
        if r == l + 1:
            return [l + 1, r + 1]
        while l < r:
            suma = numbers[l] + numbers[r] 
            if suma == target:
                break
            elif suma > target:
                r -= 1
            else:
                l += 1
        return [l + 1, r + 1]

File: Data Structures & Algorithms/two-integer-sum/submission-4.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> myMap;
        for (int i = 0; i < nums.size(); i++) {
            myMap[nums[i]] = i;
        }
        for (int i = 0; i < nums.size(); i++) {
            int dif = target - nums[i];
            if (myMap.contains(dif) && myMap[dif] != i) {
                return {i, myMap[dif]};
            }
        }
    }
};
